sdk/core/azure-core/inc/azure/core/operation.hpp

```diff
@@ -31,13 +31,47 @@ namespace Azure { namespace Core {
         = 0;
 
   protected:
+    std::unique_ptr<Azure::Core::Http::RawResponse> m_rawResponse = nullptr;
     OperationStatus m_status = OperationStatus::NotStarted;
 
+    Operation() = default;
+
+    // Define how an Operation<T> can be moved-constructed from rvalue other. Parameter `other`
+    // gave up ownership for the rawResponse.
+    Operation(Operation&& other)
+        : m_rawResponse(std::move(other.m_rawResponse)), m_status(other.m_status)
+    {
+    }
+
+    // Define how an Operation<T> can be copy-constructed from some other Operation reference.
+    // Operation will create a clone of the rawResponse from `other`.
+    Operation(Operation const& other)
+        : m_rawResponse(std::make_unique<Http::RawResponse>(other.GetRawResponse())),
+          m_status(other.m_status)
+    {
+    }
+
+    // Define how an Operation<T> can be moved-assigned from rvalue other. Parameter `other`
+    // gave up ownership for the rawResponse.
+    void operator=(Operation&& other)
+    {
+      this->m_rawResponse = std::move(other.m_rawResponse);
+      this->m_status = other.m_status;
+    }
+
+    // Define how an Operation<T> can be copy-assigned from some other Operation reference.
+    // Operation will create a clone of the rawResponse from `other`.
+    void operator=(Operation const& other)
+    {
+      this->m_rawResponse = std::make_unique<Http::RawResponse>(other.GetRawResponse());
+      this->m_status = other.m_status;
+    }
```

Comment:
why is it necessary to make `operation<T>` copiable?

Reply:
I think movable is enough.

Reply:
`operator=` should return reference to current class.

Reply:
I added to support the Storage case of creating a `Response<T>` from an `Operation<T>` without removing the rawResponse from the `Operation<T>`.  But it looks like we are OK with leaving the Operation with no raw response?

Customer Code:
auto someOperationResult = client.SomeApiReturnOperation();
auto finalValue = someOperationResult.PoolUntilDone();
// All fine up to here, but if I want to call 
auto finalValue2 = someOperationResult.PoolUntilDone(); // This won't work and throw because the rawResponse was moved out of the Operation.

Reply:
> I added to support the Storage case of creating a Response<T> from an Operation<T> without removing the rawResponse from the Operation<T>. But it looks like we are OK with leaving the Operation with no raw response?

To create a`Response<T>`, we only need to copy T and the raw response from `Operation<T>`. We don't need to copy the whole `Operation<T>` right? So I assume copy constructor and copy assignment operator of `Operation<T>` won't be necessary.

> auto finalValue2 = someOperationResult.PoolUntilDone(); // This won't work and throw because the rawResponse was moved out of the Operation.

Then let's make a copy of raw response.